<launch>

  <arg name="uav_name" default="$(optenv UAV_NAME uav3)"/>

  <arg name="standalone" default="false"/>

  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg uav_name)_uvdar_nodelet_manager"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <!-- Node Settings -->
  <arg name="output" default="screen"/>
  <arg name="calib" default="false"/>

  <arg name="DEBUG" default="false"/>
  <arg name="VisDEBUG" default="false"/>
  <arg name="gui" default="true"/>

  <arg name="frame_uvdar" default="uvcam_$(arg uav_name)"/>
  <arg name="frame_camera" default="color_camera"/>


  <arg name="offline" default="false"/>


  <arg name="threshold" default="180"/>

  <arg name="publishVisualization"         default="true"/>
  <arg name="visualizationRate"         default="1"/>

  <arg name="legacy"         default="False"/>
  <arg name="legacy_delay"         default="0.2"/>
  
 

  <arg name="device" default="$(optenv BLUEFOX_UV)"/>
  <arg name="rate" default="100"/>
  <arg name="serial" default="$(arg device)"/>
  <arg name="camera_name" default="mv_$(arg device)"/>
  <arg name="camera" default="$(arg camera_name)"/>
  <arg name="frame_id" default="$(arg camera)"/>
  <arg name="calib_url" default="file://$(find mrs_general)/camera_info/uvdar_bluefox.yaml"/>
  <arg name="fps" default="$(arg rate)"/>
  <arg name="idpf" default="0"/>
  <arg name="aec" default="false"/>
  <arg name="expose_us" default="1500"/>
  <arg name="agc" default="false"/>
  <arg name="gain_db" default="0.0"/>
  <arg name="cbm" default="0"/>
  <arg name="ctm" default="1"/>
  <arg name="dcfm" default="0"/>
  <arg name="hdr" default="false"/>
  <arg name="wbp" default="-1"/>
  <arg name="request" default="0"/>
  <arg name="mm" default="0"/>
  <arg name="jpeg_quality" default="90"/>

  <arg name="processRate" default="10"/>

  <arg name="accumulatorLength"         default="23"/>
  <arg name="pitchSteps"         default="16"/>
  <arg name="yawSteps"         default="8"/>
  <arg name="maxPixelShift"         default="2"/>

  <arg name="filterDistLength"         default="10"/>
  <arg name="filterOrientationLength"         default="10"/>

  <!-- Node Settings -->
  <arg name="view" default="false"/>

  <!-- <arg name="gui" default="false"/> -->

  <arg name="trajectoryControl" default="true"/>

  <arg name="followDistance" default="5"/>
  <arg name="trajCoeff" default="0.8"/>
  <arg name="yawCoeff" default="0.1"/>
  <arg name="tailingCoeff" default="1.0"/>

  <arg name="SamplePointSize" default="8"/>
  <arg name="StepSize" default="10"/>
  <arg name="publish" default="true"/>
  <arg name="useOdom" default="false"/>
  <arg name="cemeraRotated" default="false"/>
  <arg name="FromBag" default="false"/>
  <arg name="FromCamera" default="true"/>
  <arg name="FromVideo" default="false"/>
  <arg name="VideoNumber" default="9"/>
  <arg name="cellSize" default="64"/>
  <arg name="cellOverlay" default="16"/>
  <arg name="surroundRadius" default="3"/>
  <arg name="camNum" default="0"/>
  <arg name="useFAST" default="true"/>
  <arg name="Delay" default="100"/>
  <arg name="TimeScale" default="1.0"/>

  <arg name="calib_file_uv" default="calib_results_bf_uv_fe.txt"/>

  <arg name="mask_file" default="mask_1.bmp"/>

      <!-- <origin xyz="0.108 0 0.085" rpy="1.57079632679 4.71238898038 1.57079632679" /> -->
  <!--Usage: static_transform_publisher x y z yaw pitch roll frame_id child_frame_id  period(milliseconds)-->
<!-- OR -->
  <group ns="$(arg uav_name)">
  
      
    <!-- <node name="uvdar_uav_pose_calculator_node" pkg="uvdar" type="uvdar_uav_pose_calculator_node" output="screen" launch-prefix="debug_roslaunch"> --> 
    <node name="uvdar_uav_pose_calculator_node" pkg="uvdar" type="uvdar_uav_pose_calculator_node" output="screen"> 
      <param name="uav_name" type = "string" value="$(arg uav_name)"/>

      <param name="DEBUG"		type="bool" value="false"/> 
      <param name="gui"             type="bool" value="$(arg gui)"/>
      <param name="publish"         type="bool" value="$(arg publish)"/>
      <param name="frequenciesPerTarget" type="int" value="2"/>
      <param name="targetCount" type="int" value="2"/>
      <param name="calib_file" type="string" value="$(arg calib_file_uv)"/>
      <param name="mask_file" type="string" value="$(arg mask_file)"/>

      <param name="legacy" type="bool" value="$(arg legacy)"/>
      <param name="legacy_delay" type="double" value="$(arg legacy_delay)"/>

      <remap from="blinkersSeen" to="/$(arg uav_name)/uvdar/blinkersSeen"/>
      <remap from="estimatedFramerate" to="/$(arg uav_name)/uvdar/estimatedFramerate"/>
      <remap from="measuredPose1" to="/$(arg uav_name)/uvdar/measuredPose1"/>
      <remap from="measuredPose2" to="/$(arg uav_name)/uvdar/measuredPose2"/>
    </node>


    <!-- <node pkg="image_transport" type="republish" name="republish_bluefox" output="screen" args="compressed in:=bluefox/image_raw raw out:=bluefox2/image_raw"  /> -->
  
  <!-- <node pkg="uvdar" type="uvdar_kalman_node" name="filterUvdar" output="screen" launch-prefix="debug_roslaunch"> -->
  <!-- <node pkg="uvdar" type="uvdar_reprojector_node" name="reprojector_node" output="screen" launch-prefix="debug_roslaunch"> -->
  
  <!-- <node pkg="uvdar" type="uvdar_kalman_node" name="filterUvdar" output="screen" launch-prefix="debug_roslaunch"> -->
  <!-- <node pkg="uvdar" type="uvdar_kalman_node" name="filterUvdar" output="screen"> -->
  <!--   <param name="output_frame" value="$(arg frame_uvdar)"/> -->

  <!--   <remap from="measuredPose1" to="/$(arg uav_name)/uvdar/measuredPose1"/> -->
  <!--   <remap from="measuredPose2" to="/$(arg uav_name)/uvdar/measuredPose2"/> -->
  <!--   <remap from="filteredPose1" to="/$(arg uav_name)/uvdar/filteredPose1"/> -->
  <!--   <remap from="filteredPose2" to="/$(arg uav_name)/uvdar/filteredPose2"/> -->

  <!-- </node> -->
    </group>


  <!-- </group> -->



  </launch>
